merkletree.go

```diff
@@ -659,106 +648,18 @@ func CircomSiblingsFromSiblings(siblings []*Hash, levels int) []*Hash {
 	return siblings
 }
 
-// CircomProcessorProof defines the ProcessorProof compatible with circom. Is
-// the data of the proof between the transition from one state to another.
-type CircomProcessorProof struct {
-	OldRoot  *Hash   `json:"oldRoot"`
-	NewRoot  *Hash   `json:"newRoot"`
-	Siblings []*Hash `json:"siblings"`
-	OldKey   *Hash   `json:"oldKey"`
-	OldValue *Hash   `json:"oldValue"`
-	NewKey   *Hash   `json:"newKey"`
-	NewValue *Hash   `json:"newValue"`
-	IsOld0   bool    `json:"isOld0"`
+// TransactionInfo defines information about change merkletree.
+type TransactionInfo struct {
+	OldRoot  *Hash
+	NewRoot  *Hash
+	Siblings []*Hash
+	OldKey   *Hash
+	OldValue *Hash
+	NewKey   *Hash
+	NewValue *Hash
+	IsOld0   bool
```

Comment:
`IsOld0` and `Fnc` seem to duplicate each other. More over `IsOld0` is used in proofs but for absolutely different purpose so the naming may confuse others.
Consider get rid of `IsOld0` at all here.

Reply:
I agree, need to investigate.

Reply:
Conclusion of discussion with @AndriianChestnykh and @OBrezhniev 
`IsOld0` and `Fnc` values that do not correlate with each other